spring-integration-core/src/main/java/org/springframework/integration/channel/AbstractPollableChannel.java

```diff
@@ -99,8 +102,14 @@ public long getReceiveErrorCountLong() {
 				getMetrics().afterReceive();
 				counted = true;
 			}
-			message = interceptorList.postReceive(message, this);
+			if (message != null && logger.isDebugEnabled()) {
```

Comment:
This isn't good `if` condition if face of the next `else if`: Message can be not `null`, but `debug` can be disabled